woocommerce/woocommerce-android
# [Backend Receipts] Abstract old stripe and the new way of retriving receipts based on WC version

Closes: #10612


### Description

The PR adds fetching of backend-generated receipts starting from a specific version (the exact version is still unknown and will be changed in the following PRs) of the WC plugin. The backend-generated receipts are available for **all paid** receipts, not only the ones paid with the card on the current device in our current implementation

### Testing instructions

To test backend-generated receipts (notice that the receipts itself are not final yet) use the shared site
* Open any paid order and notice the "see receipt" button -> click and take a look at the receipt
* Create order and make card payment -> Notice that at the end of the flow, a new backend-generated receipt can be viewed

To test that there is no regression:
* Use any site with the release version of WC
* Notice that receipts only paid with a card from the given device will be provided

### Images/gif


With the support of the backend-generated receipts

https://github.com/woocommerce/woocommerce-android/assets/4923871/b4d125c1-5004-4beb-82b8-83bf58ef32a9


Without the support of the backend-generated receipts

https://github.com/woocommerce/woocommerce-android/assets/4923871/3ff0ac40-a693-4364-a859-fe819f16e961
